### PR TITLE
Prevent segfault on Python 2.7 AlignedSegment.compare(other=None)

### PR DESCRIPTION
**Pull request by rjlasko**
Comparing an `AlignedSegment` object to `None` throws a Segmentation Fault, when running on Python 2.7.  Using Python 3 did not yield the segfault in a one-off test.

The specific condition to throw the segfault is when testing object equality.  Instance equality seems to work fine.
```
a = pysam.AlignedSegment()
b = None

a is b   # False
a == b # segfault
a.compare(b) # segfault
```

The submitted PR includes additions to the unit tests to reproduce the segfault, as well as the fix.  The current workaround is for users to use the `is` operator when `None` is a potential comparison argument.

As stated before, this issue was reliably reproduced on the following, using Python 2.7.

**Docker images**
- `python:2.7.17-slim-buster`
- `python:2.7.17-alpine`
- `centos:7.5.1804`

**Bare Metal & KVM**
- CentOS 7

In the case of CentOS 7, I tried installing with pip, conda, and building from source.  None of these approaches seemed to resolve the issue.  I was unable to reproduce the bug on the `python:3.7.5-slim-buster` Docker image, so I assume that somehow Python 3 is safe.

### CODE REVIEW

File: tests/AlignedSegment_test.py [hunk 1]
```diff
@@ -78,6 +78,12 @@ def testSettingTagInEmptyRead(self):
     def testCompare(self):
         '''check comparison functions.'''
         a = self.build_read()
+        b = None
+
+        self.assertFalse(a is b)
```
Comment (rjlasko) [inline tests/AlignedSegment_test.py]: This works without the fix in place.

File: tests/AlignedSegment_test.py [hunk 2]
```diff
@@ -78,6 +78,12 @@ def testSettingTagInEmptyRead(self):
...
+        self.assertFalse(a == b)
+        self.assertFalse(a.compare(b))
```
Comment (rjlasko) [inline tests/AlignedSegment_test.py]: These two operations segfaulted immediately, and terminated the running process hard.  No other unit tests or code runs after this error is encountered.

File: pysam/libcalignedsegment.pyx [hunk 3]
```diff
@@ -999,6 +999,10 @@ cdef class AlignedSegment:
         <,=,> to *other*
         '''
 
+        # avoid segfault when other equals None
+        if other is None:
+            return -1
```
Comment (rjlasko) [inline pysam/libcalignedsegment.pyx]: Im not certain what the right return value should have been here, but `__richcmp__` (the caller that I saw in this file) seems like it doesn't care as long as it is non-zero.

Comment (kevinjacobs-progenity) [reply pysam/libcalignedsegment.pyx]: I'm looking into this now.

**Comment by AndreasHeger** [comment]
Thanks, apologies for the delay in merging.

**Pull request merged**